packages/sns/test/consumers/SnsSqsPermissionConsumer.spec.ts

```diff
@@ -435,4 +441,71 @@ describe('SnsSqsPermissionConsumer', () => {
       })
     })
   })
+
+  describe('visibility timeout', () => {
+    const topicName = 'myTestTopic'
+    const queueName = 'myTestQueue'
+    let diContainer: AwilixContainer<Dependencies>
+
+    beforeEach(async () => {
+      diContainer = await registerDependencies({
+        permissionConsumer: asFunction(() => undefined),
+        permissionPublisher: asFunction(() => undefined),
+      })
+    })
+
+    afterEach(async () => {
+      await diContainer.cradle.awilixManager.executeDispose()
+      await diContainer.dispose()
+    })
+
+    it.each([false, true])('using 2 consumers with heartbeat -> %s', async (heartbeatEnabled) => {
+      let consumer1IsProcessing = false
+      let consumer1Counter = 0
+      let consumer2Counter = 0
+
+      const consumer1 = new SnsSqsPermissionConsumer(diContainer.cradle, {
+        creationConfig: {
+          topic: { Name: topicName },
+          queue: { QueueName: queueName, Attributes: { VisibilityTimeout: '2' } },
+        },
+        consumerOverrides: { heartbeatInterval: heartbeatEnabled ? 1 : undefined },
+        removeHandlerOverride: async () => {
+          consumer1IsProcessing = true
+          // wait for consumer2 to process message while this is still processing
+          await waitAndRetry(() => consumer2Counter > 0, 100, 30)
```

Comment:
we probably should also store the result of this `waitAndRetry` and assert it resolved to undefined, if intention is to ensure it never processes. or the idea is that it just takes long? then we should assert it returns true :D

Reply:
At this point, I am only forcing this wait to give time to consumer2 to process the message, depending on the test it will resolves to `true` or `false`, I am checking it at the end of the test by using the counter.

The reason why I don't check it here is because I prefer to have the Given-When-Then structure and also to reduce the number of `heatbeatIntervalEnabled` checks in the test

Reply:
if we don't care about the actual condition, maybe we should just do `await setTimeout(fixedTime)` (from `timers/promises`) then?

Reply:
Fair point! I will modify it